scripts/scripts.js

```diff
@@ -90,9 +90,65 @@ sampleRUM.observe = ((elements) => {
   });
 });
 
+sampleRUM.targetselector = (element) => {
+  let selector = element.tagName.toLowerCase();
+  // always include these attributes as they have valuable info
+  const distinctAttributes = ['src', 'href'];
+  // capture the actual click target instead of a child element
+  const terminateearly = (e) => e.tagName.toLowerCase() === 'a'
+    || e.tagName.toLowerCase() === 'button';
+  if (element === document.body || element === document.documentElement) {
+    return selector;
+  }
+  if (element.id) {
+    return `#${element.id}`;
+  }
+  if (element.className) {
+    selector = `${selector}.${Array.from(element.classList).join('.')}`;
+  }
+  selector = distinctAttributes.reduce((pv, attname) => {
+    if (element.getAttribute(attname)) {
+      return `${pv}[${attname}="${element.getAttribute(attname)}"]`;
+    }
+    return pv;
+  }, selector);
+  // check for uniqueness
+  const otherchildren = element.parentElement.querySelectorAll(element.tagName);
+  for (let i = 0; (
+    i < otherchildren.length
+    && otherchildren.length > 1
+    && element.parentElement.querySelectorAll(selector).length > 1);
+    i += 1) {
+    if (otherchildren[i] === element) {
+      selector = `${element.tagName.toLowerCase()}:nth-of-type(${i + 1})`;
```

Comment:
If there are multiple matching elements, we use `:nth-of-type()` to clarify, which one has been clicked:

```
main > div.section-wrapper.recommended-articles-small-container > div.recommended-articles-small-content-wrapper > p:nth-of-type(1)
```